spot-contracts/contracts/_utils/Sigmoid.sol

```diff
@@ -32,29 +70,25 @@ library Sigmoid {
     ) internal pure returns (int256) {
         int256 delta;
 
-        delta = (x.sub(one));
+        delta = x - one;
 
         // Compute: (Upper-Lower)/(1-(Upper/Lower)/2^(Growth*delta))) + Lower
 
-        int256 exponent = growth.mul(delta).div(one);
+        int256 exponent = (growth * delta) / one;
+
         // Cap exponent to guarantee it is not too big for twoPower
-        if (exponent > one.mul(100)) {
-            exponent = one.mul(100);
-        }
-        if (exponent < one.mul(-100)) {
-            exponent = one.mul(-100);
-        }
+        exponent = SignedMathUpgradeable.min(SignedMathUpgradeable.max(exponent, -100 * one), 100 * one);
```

Comment:
Is this clearer than the original version? I think I find it easier to understand the conditional form. I'm guessing it's also more gas efficient, since there are no jumps.

Reply:
sure will go back